source/extensions/tracers/datadog/time_util.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <datadog/clock.h>
+
+#include "envoy/common/time.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+
+// Return the `datadog::tracing::TimePoint` that most closely matches the
+// specified `SystemTime`. Use the optionally specified
+// `datadog::tracing::Clock` to read the current time. If a clock is not
+// specified, then the default clock is used.
+datadog::tracing::TimePoint estimateTime(SystemTime);
```

Comment:
is this an *estimate* really? or are we just converting a SystemTime to a DataDog time?

Would it be clearer at call-sites to call this convertSystemTimeToDataDog() ?

Reply:
Yes, it's an estimate. The background is explained [here][1] as part of the original pull request (that readme will be added again in a subsequent pull request).

Envoy has a time type that, like Go's, contains both a system time point and a steady ("monotonic") time point. However, only the system time is exposed to the tracing subsystem. Not sure why.

This is problematic for the new library (dd-trace-cpp), because it uses the steady time to calculate the duration of a span (`end.tick - begin.tick`). So, we need to get a steady clock time from a given system clock time. The scheme is to measure the current system/steady time, compare the system part with the given system time, and then adjust the measured steady time accordingly. This is correct if the system clock has not been adjusted since the given system time was measured. It's incorrect otherwise, hence an estimate.

What do you think about all this? Is there a better way?

[1]: https://github.com/DataDog/envoy/blob/david.goffredo/dd-trace-cpp/source/extensions/tracers/datadog/README.md#time_utilhcc

Reply:
sgtm thanks. The above explanation might be good to throw into the code here as a comment.